retroshare-gui/src/gui/settings/WebuiPage.cpp

```diff
@@ -181,15 +191,21 @@ void WebuiPage::onAllIPCBClicked(bool /*checked*/)
 	QString S;
 	updateParams(S);
 }
-
 void WebuiPage::onApplyClicked()
 {
-    QString errmsg;
-    bool ok = updateParams(errmsg);
-    if(!ok)
+    rsWebUI->setUserPassword(ui.password_LE->text().toStdString());
+
+    if(!restart())
     {
         QMessageBox::warning(0, tr("failed to start Webinterface"), "Failed to start the webinterface.");
         return;
     }
-    QDesktopServices::openUrl(QUrl(QString("http://localhost:")+QString::number(ui.port_SB->value())));
+
+    emit passwordChanged();
 }
+
+void WebuiPage::onStartWebBrowserClicked()
+{
+    QDesktopServices::openUrl(QUrl(QString("http://localhost:")+QString::number(rsJsonAPI->listeningPort())));

```

Comment:
localhost may be traslateb both to `127.0.0.1` and to `::1` in our case restbed either bind on one or the other so depending on host configuration opening that url may fail, so please change `localhost` with the correct bind address, which moreover could be configured to something else